src/main/java/org/apache/commons/collections4/bloomfilter/hasher/package-info.java

```diff
@@ -16,10 +16,64 @@
  */
 
 /**
- * Provides classes and interfaces to define the shape of a Bloom filter and the conversion
- * of generic bytes to a hash of bit indexes to be used with a Bloom filter.
+ * Hasher implementations and tools.
  *
+ *
+ * <h2>Hasher</h2>
+ *
+ * <p>A Hasher converts one or more items into an {@code IndexProducer} series of integers based on a {@code Shape}.
+ *
+ *
+ * The base Hasher implementations
+ * are as follows:</p>
+ *
+ *  <h3>SimpleHasher</h3>
+ *
+ *  <p>The SimpleHasher represents one item being added to the Bloom filter.  It utilizes the combinatorial strategy
+ *  as described by <a href='https://www.eecs.harvard.edu/~michaelm/postscripts/tr-02-05.pdf'>Krisch and Mitzenmacher</a>.
+ *  Generally, a hash value is created by hashing together multiple properties of the item being added. The hash value is
+ *  then used to create a SimpleHasher.</p>
+ *
+ *  <p>This hasher represents a single item and thus does not return duplicate indices.</p>
+ *
+ *  <h3>HasherCollection</h3>
+ *
+ *  <p>The HasherCollection is a collection of Hashers that implements the Hasher interface.  Each hasher within the collection
+ *  represents a single item, or in the case of a HasherCollections multiple items.</p>
+ *
+ *  <p>This hahser represents multiple items and thus may return duplicate indices.</p>
+ *
+ *  <h3>SingleItemHasherCollection</h3>
+ *
+ *  <p>A collection of Hashers that are combined to represent a single item.  Like the HasherCollection this Hasher is composed
+ *  of multiple Hashers.  Unlike the HasherCollection, this hasher reports that it is only one item.</p>
+ *
+ *
+ *  <p>This hasher represents a single item and thus does not return duplicate indices.</p>
+ *
+ *  <h3>Other Implementations</h3>
+ *
+ *  <p>Other implementations of the Hasher are easy to implement.  Hashers that represent single items should make use of the
+ *  {@code Hahser.Filter} and/or {@code Hahser.FileredIntConsumer} classes to filter out duplicate indices.</p>
```

Comment:
s/Hahser/Hasher twice (a misspelling easy to happen, worth checking in the rest of the code/docs with `grep` or IDE)